Mathlib/Analysis/Meromorphic/NormalFormAt.lean

```diff
@@ -370,6 +374,270 @@ theorem meromorphicNFAt_toMeromorphicNFAt :
+    toMeromorphicNFOn f U = 0 := by
+  simp [toMeromorphicNFOn, hf]
+
+/--
+Conversion to normal form on `U` does not change values outside of `U`.
+-/
+@[simp] lemma toMeromorphicNFOn_eq_self_on_compl (hf : MeromorphicOn f U) :
+    Set.EqOn f (toMeromorphicNFOn f U) Uᶜ := by
+  intro x hx
+  simp_all [toMeromorphicNFOn]
+
+/--
+Conversion to normal form on `U` changes the value only along a discrete subset
+of `U`.
+-/
+theorem toMeromorphicNFOn_eqOn_codiscrete [CompleteSpace E] (hf : MeromorphicOn f U) :
+    f =ᶠ[Filter.codiscreteWithin U] toMeromorphicNFOn f U := by
+  have : U ∈ Filter.codiscreteWithin U := by
+    simp [mem_codiscreteWithin.2]
+  filter_upwards [hf.analyticAt_mem_codiscreteWithin, this] with a h₁a h₂a
+  simp [toMeromorphicNFOn, hf, ← toMeromorphicNFAt_eq_self.2 h₁a.meromorphicNFAt]
+
+/--
+If `f` is meromorphic on `U` and `x ∈ U`, then `f` and its conversion to normal
+form on `U` agree in a punctured neighborhood of `x`.
+-/
+theorem MeromorphicOn.toMeromorphicNFOn_eq_self_on_nhdNE [CompleteSpace E]
+    (hf : MeromorphicOn f U) (hx : x ∈ U) :
+    toMeromorphicNFOn f U =ᶠ[𝓝[≠] x] f := by
+  filter_upwards [(hf x hx).eventually_analyticAt] with a ha
+  simp [toMeromorphicNFOn, hf, ← toMeromorphicNFAt_eq_self.2 ha.meromorphicNFAt]
+
+/--
+If `f` is meromorphic on `U` and `x ∈ U`, then conversion to normal form at `x`
+and conversion to normal form on `U` agree in a neighborhood of `x`.
+-/
+theorem toMeromorphicNFOn_eq_toMeromorphicNFAt_on_nhd [CompleteSpace E] (hf : MeromorphicOn f U)
+    (hx : x ∈ U) :
+    toMeromorphicNFOn f U =ᶠ[𝓝 x] toMeromorphicNFAt f x := by
+  apply eventuallyEq_nhds_of_eventuallyEq_nhdsNE
+  · exact (hf.toMeromorphicNFOn_eq_self_on_nhdNE hx).trans (hf x hx).eq_nhdNE_toMeromorphicNFAt
+  · simp [toMeromorphicNFOn, hf, hx]
+
+/--
+If `f` is meromorphic on `U` and `x ∈ U`, then conversion to normal form at `x`
+and conversion to normal form on `U` agree at `x`.
+-/
+theorem toMeromorphicNFOn_eq_toMeromorphicNFAt [CompleteSpace E] (hf : MeromorphicOn f U)
+    (hx : x ∈ U) :
+    toMeromorphicNFOn f U x = toMeromorphicNFAt f x x := by
+  apply Filter.EventuallyEq.eq_of_nhds (g := toMeromorphicNFAt f x)
+  simp [(toMeromorphicNFOn_eq_toMeromorphicNFAt_on_nhd hf hx).trans]
+
+variable (f U) in
+/--
+After conversion to normal form on `U`, the function has normal form.
+-/
+theorem meromorphicNFOn_toMeromorphicNFOn [CompleteSpace E] :
+    MeromorphicNFOn (toMeromorphicNFOn f U) U := by
+  by_cases hf : MeromorphicOn f U
+  · intro z hz
+    rw [meromorphicNFAt_congr (toMeromorphicNFOn_eq_toMeromorphicNFAt_on_nhd hf hz)]
     exact meromorphicNFAt_toMeromorphicNFAt
+  · simp [hf]
+    apply AnalyticOnNhd.meromorphicNFOn
+    exact analyticOnNhd_const
+
+/--
+If `f` has normal form on `U`, then `f` equals `toMeromorphicNFOn f U`.
+-/
+@[simp] theorem toMeromorphicNFOn_eq_self [CompleteSpace E] :
+    MeromorphicNFOn f U ↔ f = toMeromorphicNFOn f U := by
```

Comment:
```suggestion
    toMeromorphicNFOn f U = f ↔ MeromorphicNFOn f U := by
```

Reply:
Done.

Reply:
Looks like you didn't quite follow my suggestion though. You seem to have done
```suggestion
    f = toMeromorphicNFOn f U ↔ MeromorphicNFOn f U := by
```
instead

Reply:
Ah!

Reply:
Fixed.